### PR TITLE
🎉 Declare Oauth init parameters for 4 connector

### LINKED ISSUE #5633: Source FB marketing: Declare oauth parameters
## Tell us about the problem you're trying to solve
blocked on #5632 

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### LINKED ISSUE #5635: Source Salesforce: declare oauth parameters in spec
## Tell us about the problem you're trying to solve
blocked on #5632

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### LINKED ISSUE #6166: Annotate Oauth init parameters for initial Cloud release
## Tell us about the problem you're trying to solve
We need to do the equivalent of [this PR](https://github.com/airbytehq/airbyte/pull/6022) for any oauth connectors: 

- #5635 
- #5634 
- #5633 
- instagram
- facebook pages

See the description of this param [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L211)

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
Annotate Oauth init parameters for connectors: Facebook Marketing/Pages, Instagram, Salesforce

## What
*closes #6166 (except for Google Connectors).*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json [hunk 1]
```diff
@@ -50,5 +50,11 @@
     "required": ["account_id", "access_token", "start_date"]
   },
   "supportsIncremental": true,
-  "supported_destination_sync_modes": ["append"]
+  "supported_destination_sync_modes": ["append"],
+  "authSpecification": {
+    "auth_type": "oauth2.0",
+    "oauth2Specification": {
+      "oauthFlowInitParameters": [["access_token"]]
```
Comment (avida) [inline airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:57]: it should be `[["access_token"], ["access_token"]]`.
You can read about oauthFlowInitParameters here:
https://github.com/airbytehq/airbyte/blob/3378ba430feb5d994ccda71821de89382f37e7a0/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L216

**Review by avida — CHANGES_REQUESTED** [review]
~~Please update all the  oauthFlowInitParameters  according to reference https://github.com/airbytehq/airbyte/blob/3378ba430feb5d994ccda71821de89382f37e7a0/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L216~~ You commit is ok, my bad

**Review by avida — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]
btw, why there are no changes for google connectors as mentioned here #5634?
`Google Analytics` is covered by #6414, but what about `Google Ads` and `Google Search console`?

**Comment by yevhenii-ldv** [comment]
@Zirochkaa Google Ads is [here](https://github.com/airbytehq/airbyte/pull/6022). As for the rest of Google connectors, @gaart took over this issue, so I did not touch them.

**Comment by ChristopheDuong** [comment]
Beware, we're having to slightly modify the oauth specs in this PR:
https://github.com/airbytehq/airbyte/pull/6456

So connectors will have to be tweaked accordingly

**Comment by ChristopheDuong** [comment]
In the case of facebook connectors, see this comment too: https://github.com/airbytehq/airbyte/issues/5989#issuecomment-927873341

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json (hunk 1 above)
Comment (sherifnada) [inline airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:57]: this should be updated as a output parameter using the new spec

**Review by sherifnada — COMMENTED** [review]
@yevhenii-ldv could you update this to the new spec format?

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json]: @sherifnada Done, could u review this PR one more time?

**Review by sherifnada — APPROVED** [review]

**Pull request merged**